forem/forem
# Show friendly error message on API fetch failure on stats

Signed-off-by: Takuya Noguchi <takninnovationresearch@gmail.com>

## What type of PR is this? (check all applicable)


## Description

Shows friendly error message on API fetch failures on `/:user/:slug/stats`.

## Related Tickets & Documents

- Related Issue #13804

## QA Instructions, Screenshots, Recordings

1. Install uBlock Origin on your Chrome browser: https://chrome.google.com/webstore/detail/ublock-origin/cjpalhdlnbpafiamejdnhcphjbkeiagm?hl=en
2. Make sure EasyPrivacy is enabled as uBlock Origin filter list: https://easylist.to/
3. Prepare Rails app with this PR to run Rails server.
4. Access to `/orville_rippin/a-summer-bird-cage-quibusdam-impedit-47p9/stats` on your dev server.

| before | after |
|-|-|
| ![Screenshot 2021-05-19 at 16 45 44](https://user-images.githubusercontent.com/5831347/118833419-b6341e80-b8c1-11eb-9630-a632445a7203.png) adapted from #13804 (courtesy of @TastefulElk) | ![](https://user-images.githubusercontent.com/10229505/166728338-93eed757-5849-460e-a3d6-69543bb78b9b.png) (768px screen width) |

cf. Another screenshot (with 1024px screen width):
![3000-forem-forem-pgkt86igaqj ws-us43 gitpod io_orville_rippin_a-summer-bird-cage-quibusdam-impedit-47p9_stats(iPad Mini)](https://user-images.githubusercontent.com/10229505/166728354-aca229de-5398-4460-a6c0-28bc880d2041.png)

Although I know the team is getting rid of Bootstrap CSS (#17412), Bootstrap is used additionally in this PR as it is already used on this page.

### UI accessibility concerns?
n/a

## Added/updated tests?

- [ ] No, and this is why: there are no tests for app/javascript/analytics/client.js and app/javascript/analytics/dashboard.js at all.
